src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -103,6 +98,41 @@ void SetLastWriteTime()
 
             try
             {
+                RetryUtil.RetryOnIOError(() =>
+                {
+                    MemoryMappedFile memoryMappedFile = null;
+                    MemoryMappedViewAccessor memoryMappedViewAccessor = null;
+                    try
+                    {
+                        // Open the source host file.
+                        memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostSourceFilePath, FileMode.Open, null, 0, MemoryMappedFileAccess.CopyOnWrite);
+                        memoryMappedViewAccessor = memoryMappedFile.CreateViewAccessor(0, 0, MemoryMappedFileAccess.CopyOnWrite);
+
+                        // Transform the host file in-memory.
+                        RewriteAppHost(memoryMappedViewAccessor);
+
+                        // Save the transformed host.
+                        BinaryUtils.SaveFile(memoryMappedViewAccessor, appHostDestinationFilePath);
+                    }
+                    finally
+                    {
+                        if (memoryMappedViewAccessor != null)
+                        {
+                            memoryMappedViewAccessor.Dispose();
+                        }
+                        if (memoryMappedFile != null)
+                        {
+                            memoryMappedFile.Dispose();
+                        }
+                    }
+                });
+
+                RetryUtil.RetryOnWin32Error(UpdateResources);
+
+                RetryUtil.RetryOnIOError(RemoveSignatureIfMachO);
+
+                RetryUtil.RetryOnIOError(SetLastWriteTime);
```

Comment:
Is this necessary if we're calling `SaveFile`? It looks like the previous concern was that simple memory map modification didn't update the last write time, but I'm unclear if that's also true of `SaveFile`.

Reply:
Good point.  It looks like the last write time is updated when we open and write the stream, so I've removed this call.

Reply:
@brianrob Any comment on this?

Reply:
Never mind, the diff here just didn't reparent after your changes.